src/libinit.js

```diff
@@ -126,7 +137,8 @@ export const LibraryInitGeneratorMixin = (B) => class extends B {
 	_handlePrompts(data) {
 		Object.assign(this.options, data);
 		if (this.options.name) {
-			this.options.Name = capitalizeFirstLetter(this.options.name);
+			this.options.name_code = lowercaseFirstLetter(this.options.name.replace(/[^a-zA-Z0-9_]/g, ''));
```

Comment:
I feel the name should be validated and an error reported so the user has a chance to fix it, rather than silently changing it.